python/pyarrow/_fs.pyx

```diff
@@ -335,6 +448,21 @@ cdef class FileSystem(_Weakrefable):
         tuple of (FileSystem, str path)
             With (filesystem, path) tuple where path is the abstract path
             inside the FileSystem instance.
+
+        Examples
+        --------
+        Create a FileSystem subclass with nonempty file:
+
+        >>> getfixture('local_fs')
+        <pyarrow._fs.LocalFileSystem object at ...>
+
+        Create a new FileSystem subclass from path:
+
+        >>> local_new, path = fs.LocalFileSystem().from_uri('/tmp/fileinfo.dat')
```

Comment:
It would probably be good to show here the use of an actual URI (the above uses an absolute local file path, so eg could use `file:///tmp/fieinfo.dat` instead for a URI, or more typically something with `s3://...`)